src/__tests__/index.ts

```diff
@@ -17,28 +17,39 @@ test('can pass one check', async () => {
   const result = await isMyFileReady(file, endsWithStr(end));
   expect(result).toStrictEqual({
     isReady: true,
-    checks: {
-      endsWithStr: {
+    checks: [
+      {
+        name: 'endsWithStr',
         isReady: true,
         endsWith: end,
       },
-    },
+    ],
   });
 });
 
 test('can pass multiple checks', async () => {
   const result = await isMyFileReady(file, [endsWithStr(end), sameSize(size)]);
   expect(result).toStrictEqual({
     isReady: true,
-    checks: {
-      endsWithStr: {
+    checks: [
+      {
+        name: 'endsWithStr',
         isReady: true,
         endsWith: end,
       },
-      sameSize: {
+      {
+        name: 'sameSize',
         isReady: true,
         size: size,
       },
-    },
+    ],
   });
 });
+
+test('typescript autocomplete correctly', async () => {
+  const result = await isMyFileReady(file, [endsWithStr(end), sameSize(size)]);
+  const check = result.checks[1];
+  if (check.name === 'endsWithStr') {
+    expect(check.endsWith).toBe(end);
```

Comment:
can you check the right array element and add `expect.assertions()` to make sure it's called?

Reply:
Done